gateway/api.go

```diff
@@ -3262,6 +3267,20 @@ func ctxSetDefaultVersion(r *http.Request) {
 	setCtxValue(r, ctx.VersionDefault, true)
 }
 
+func ctxSetSelfLooping(r *http.Request, value bool) {
```

Comment:
Move to internal/httpctx and extend godoc to cover both functions.
Rename to SetSelfLooping and IsSelfLooping.